DOC-2217: Update `ai_shortcut` documentation with `selection` property and changes to default value. (#2997)
tinymce/tinymce-docs: modules/ROOT/partials/configuration/ai_shortcuts.adoc

=== Valid Shortcuts

Valid shortcut objects contain the following properties.

`+title+`:: A string which is displayed in the `+aishortcuts+` toolbar button and menu item. This will indicate which shortcut is used, or which category of shortcuts are in this menu.

And either

`+subprompts+`:: An array containing more valid shortcut objects. This allows shortcuts to be grouped into categories within the {pluginname} shortcuts toolbar button and menu item.

or

`+prompt+`:: A string containing the query which is given to the `+ai_request+` function when the shortcut is used.

> ==== The `+selection+` property

Reviewer: Do we need to specify what the default value is?

Author: It can be undefined

include::partial$misc/admon-requires-6.8v.adoc[]

Shortcut objects with the `+prompt+` property may also contain the following optional property.

`+selection+`:: A boolean value which will match to the current selection and set the enabled state of the shortcut. When `+selection+` is:

* `+true+`: The shortcut will _only_ be enabled when the user _has_ made a selection in the editor.

* `+false+`: The shortcut will _only_ be enabled when the user _has not_ made a selection in the editor.

* `+undefined+`, or _not set_: The shortcut will always be enabled.

> This property allows you to define shortcuts with `+selection: true+` which should only operate on context, when the user has selected content. Alternatively, shortcuts with `+selection: false+` will only be used with no selection, allowing you to define shortcuts for generating content from a specific prompt, without any context.

Reviewer: ```suggestion
This property allows shortcuts with `+selection: true+` to be defined that should only operate on context when the user has selected content. Alternatively, shortcuts with `+selection: false+` will only be used with no selection, thereby allowing shortcuts to be defined for generating content from a specific prompt, without any context.
```
